analyzers/src/SonarAnalyzer.Common/Rules/Hotspots/HardcodedIpAddressBase.cs

```diff
@@ -98,6 +98,7 @@ private void CheckForHardcodedIpAddressesInStringInterpolation(SyntaxNodeAnalysi
         private static bool IsRoutableNonLoopbackIPAddress(string literalValue, out IPAddress ipAddress) =>
             IPAddress.TryParse(literalValue, out ipAddress)
             && !IPAddress.IsLoopback(ipAddress)
+            && !(ipAddress.IsIPv4MappedToIPv6 && IPAddress.IsLoopback(ipAddress.MapToIPv4()))
```

Comment:
Simplified version. See https://github.com/SonarSource/sonar-dotnet/issues/6080#issuecomment-1268625170 for details.